### ISSUE TITLE
don't report constrained `TypeVar` with variance if the constraints are assignable

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#876]
```py
T1 = TypeVar("T1", object, int, covariant=True)  # expect no error
T2 = TypeVar("T2", int, str, covariant=True)  # expect error

class A(Generic[T1]):
    def get(self) -> T: ...

a_int = A[int]()
a_obj = a_int  # valid assignment
```
